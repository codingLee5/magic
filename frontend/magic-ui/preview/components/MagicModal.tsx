import React, { useState } from "react"
import { Space, Button } from "antd"
import MagicModal from "../../components/MagicModal"
import MagicButton from "../../components/MagicButton"
import ComponentDemo from "./Container"

const ModalDemo: React.FC = () => {
	const [isModalOpen1, setIsModalOpen1] = useState(false)
	const [isModalOpen2, setIsModalOpen2] = useState(false)
	const [isModalOpen4, setIsModalOpen4] = useState(false)

	return (
		<div>
			<ComponentDemo
				title="基础模态框"
				description="最基本的模态框组件"
				code="<MagicModal open={isOpen} onCancel={() => setIsOpen(false)} />"
			>
				<Space>
					<MagicButton type="primary" onClick={() => setIsModalOpen1(true)}>
						打开基础模态框
					</MagicButton>
					<MagicModal
						title="基础模态框"
						open={isModalOpen1}
						onCancel={() => setIsModalOpen1(false)}
						onOk={() => setIsModalOpen1(false)}
					>
						<p>这是一个基础模态框的内容示例。</p>
						<p>点击确定或取消按钮可以关闭模态框。</p>
					</MagicModal>
				</Space>
			</ComponentDemo>

			<ComponentDemo
				title="不同尺寸的模态框"
				description="支持不同尺寸的模态框"
				code="width: number | string"
			>
				<Space>
					<MagicButton onClick={() => setIsModalOpen2(true)}>小尺寸模态框</MagicButton>
					<MagicModal
						title="小尺寸模态框"
						open={isModalOpen2}
						width={400}
						onCancel={() => setIsModalOpen2(false)}
						onOk={() => setIsModalOpen2(false)}
					>
						<p>这是一个宽度为400px的小尺寸模态框。</p>
					</MagicModal>
				</Space>
			</ComponentDemo>

			<ComponentDemo
				title="无按钮模态框"
				description="可以隐藏默认的确定和取消按钮"
				code="footer: null"
			>
				<Space>
					<MagicButton onClick={() => setIsModalOpen4(true)}>无按钮模态框</MagicButton>
					<MagicModal
						title="无按钮模态框"
						open={isModalOpen4}
						footer={null}
						onCancel={() => setIsModalOpen4(false)}
					>
						<p>这个模态框没有默认的按钮。</p>
						<p>只能通过点击右上角的关闭按钮或遮罩层来关闭。</p>
						<div style={{ textAlign: "right", marginTop: 16 }}>
							<Button onClick={() => setIsModalOpen4(false)}>自定义关闭按钮</Button>
						</div>
					</MagicModal>
				</Space>
			</ComponentDemo>
		</div>
	)
}

export default ModalDemo
