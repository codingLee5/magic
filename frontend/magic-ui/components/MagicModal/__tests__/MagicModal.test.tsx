import { render, screen } from "@testing-library/react"
import MagicThemeProvider from "../../ThemeProvider"
import { describe, it, expect } from "vitest"
import MagicModal from "../index"

const renderWithTheme = (component: React.ReactElement) =>
	render(<MagicThemeProvider theme="light">{component}</MagicThemeProvider>)

describe("MagicModal", () => {
	it("应该正常渲染", () => {
		renderWithTheme(
			<MagicModal open title="测试标题">
				测试内容
			</MagicModal>,
		)
		expect(screen.getByText("测试标题")).toBeInTheDocument()
		expect(screen.getByText("测试内容")).toBeInTheDocument()
	})

	// 快照测试
	describe("快照测试", () => {
		it("基础模态框快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicModal open title="基础模态框">
					这是模态框内容
				</MagicModal>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("关闭状态模态框快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicModal open={false} title="关闭的模态框">
					这是模态框内容
				</MagicModal>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带确认按钮模态框快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicModal open title="确认模态框" okText="确认" cancelText="取消">
					确认要执行此操作吗？
				</MagicModal>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("自定义宽度模态框快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicModal open title="宽模态框" width={800}>
					这是一个较宽的模态框
				</MagicModal>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("无标题模态框快照", () => {
			const { asFragment } = renderWithTheme(<MagicModal open>没有标题的模态框</MagicModal>)
			expect(asFragment()).toMatchSnapshot()
		})

		it("居中模态框快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicModal open title="居中模态框" centered>
					这是一个居中的模态框
				</MagicModal>,
			)
			expect(asFragment()).toMatchSnapshot()
		})
	})
})
