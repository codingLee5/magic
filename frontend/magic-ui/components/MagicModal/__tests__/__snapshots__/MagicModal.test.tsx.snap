// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MagicModal > 快照测试 > 关闭状态模态框快照 1`] = `<DocumentFragment />`;

exports[`MagicModal > 快照测试 > 基础模态框快照 1`] = `<DocumentFragment />`;

exports[`MagicModal > 快照测试 > 居中模态框快照 1`] = `<DocumentFragment />`;

exports[`MagicModal > 快照测试 > 带确认按钮模态框快照 1`] = `<DocumentFragment />`;

exports[`MagicModal > 快照测试 > 无标题模态框快照 1`] = `<DocumentFragment />`;

exports[`MagicModal > 快照测试 > 自定义宽度模态框快照 1`] = `<DocumentFragment />`;
